README.md

```diff
@@ -14,6 +14,7 @@ dotnet add package JsonApiDotNetCore.MongoDb
 ### Models
 
 ```cs
+// MongoDbIdentifiable is just a utility base class, could use  IIdentifiable<TId> instead
```

Comment:
This is not true. Today we have at least one strong dependency on it [here](https://github.com/json-api-dotnet/JsonApiDotNetCore.MongoDb/blob/08b6231649d6d7d5119af6e42f2d4bd3e12e3cf9/src/JsonApiDotNetCore.MongoDb/Serialization/Building/IgnoreRelationshipsResponseResourceObjectBuilder.cs#L28), with more to follow in the future in cases we need to know whether we are dealing with MongoDB.

Reply:
Clearly had not appreciated this, thanks. Will remove.

Have a personal preference for fluent style persistence mappers where no need to inherit from base classes/use attributes when using frameworks to avoid leaking http/api semantics over business logic which was why I was minimising it here, but accept that is not supported with json-api-dotnet at this time.

Reply:
I get your point and I agree. But the thing is, JADNC wasn't designed that way. We have several open issues on that and intend to rethink such separation in a [future version](https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/ROADMAP.md). Doing that now would be very breaking, so it is not an option for JADNC v4.x. The MongoDB implementation heavily depends on query building logic that was designed for EF Core, so it cannot function without a reference to EF Core currently. All-in-all, there are quite some hidden dependencies we cannot easily get rid of.